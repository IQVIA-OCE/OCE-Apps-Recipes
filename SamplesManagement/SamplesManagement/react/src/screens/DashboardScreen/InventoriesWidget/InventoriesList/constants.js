export const inventoriesColors = {
  AdHocInventory: '#b781d3',
  AuditedInventory: '#4ad463',
  InitialInventory: '#769ed9',
  PeriodicInventory: '#ef6e64',
}
