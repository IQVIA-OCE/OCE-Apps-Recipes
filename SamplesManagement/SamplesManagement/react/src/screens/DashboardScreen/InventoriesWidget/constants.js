export const inventoriesColors = {
  AdHocInventory: '#b781d3',
  AuditedInventory: '#4ad463',
  InitialInventory: '#769ed9',
  PeriodicInventory: '#ef6e64',
}

export const INVENTORY_TYPE = {
  InitialInventory: 'Initial Inventory',
  AdHocInventory: 'Ad Hoc Inventory',
  AuditedInventory: 'Audited Inventory',
  PeriodicInventory: 'Periodic Inventory',
};