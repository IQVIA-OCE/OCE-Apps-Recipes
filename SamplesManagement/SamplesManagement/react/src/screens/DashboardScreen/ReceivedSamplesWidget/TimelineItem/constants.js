export const recordTypes = {
  AcknowledgementOfShipment: {
    icon: 'format-list-checks',
    color: '#ef6e64',
  },
  TransferIn: {
    icon: 'format-indent-increase',
    color: '#f88962',
  },
};
