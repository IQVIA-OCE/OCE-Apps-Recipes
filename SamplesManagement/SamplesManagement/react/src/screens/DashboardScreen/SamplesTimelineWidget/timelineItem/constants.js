export const recordTypes = {
  AcknowledgementOfShipment: {
    icon: 'format-list-checks',
    color: '#ef6e64',
  },
  Adjustment: {
    icon: 'tune-vertical',
    color: '#47cfd2',
  },
  Disbursement: {
    icon: 'file-document-box-outline',
    color: '#769ed9',
  },
  TransferIn: {
    icon: 'format-indent-increase',
    color: '#f88962',
  },
  TransferOut: {
    icon: 'format-indent-decrease',
    color: '#f88962',
  },
  Order: {
    icon: 'clipboard-check',
    color: '#69cd37',
  },
  Return: {
    icon: 'package-variant-closed',
    color: '#b781d3',
  },
};
