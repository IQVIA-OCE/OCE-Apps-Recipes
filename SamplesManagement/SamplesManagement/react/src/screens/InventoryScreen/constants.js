export const DATE_TIME_FORMAT = 'MMM D, YYYY h:mm A';
export const DATE_TIME_FORMAT_WITH_SECONDS = 'MMM D, YYYY h:mm:ss A';
export const DATE_FORMAT = 'MMM D, YYYY';
export const INVENTORY_STATUS = {
  inProgress: 'In Progress',
  saved: 'Saved',
  submitted: 'Submitted',
};

export const BANNER_CONFIG = {
  inProgress: {
    variant: 'success',
    message: 'In Progress. Saved Successfully',
    visible: true,
    icon: 'checkbox-marked-circle',
  },
  saved: {
    variant: 'warning',
    message: 'Saved. Need reason for discrepancies',
    visible: true,
    icon: 'alert',
  },
  submitted: {
    variant: 'success',
    message: 'Submitted Successfully',
    visible: true,
    icon: 'checkbox-marked-circle',
  },
};

export const REASON_LIST = [
  { label: 'Theft/Loss', id: 'Theft/Loss' },
  { label: 'Damaged', id: 'Damaged' },
  { label: 'Disbursement Error', id: 'Disbursement Error' },
  { label: 'Other', id: 'Other' },
];
