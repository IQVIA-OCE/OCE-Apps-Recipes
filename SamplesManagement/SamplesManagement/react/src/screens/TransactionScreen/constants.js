export const packageConditions = [
  { label: 'Undamaged', id: '1' },
  { label: 'Damaged', id: '2' },
  { label: 'Opened', id: '3' },
];
