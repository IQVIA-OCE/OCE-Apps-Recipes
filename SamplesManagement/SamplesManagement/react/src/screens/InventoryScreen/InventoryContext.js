import React, { useContext, useEffect, useState } from 'react';
import {
  fetchActiveLotsProducts,
  fetchConfigs,
  fetchInventoryById,
  fetchInventoryDetail,
  fetchInventoryTypes,
  fetchLastSubmittedInventory,
  fetchTransactionDetails,
} from '../../api/Inventories';
import { normalizer } from '../../utils/utils';
import { AppContext } from '../../../AppContext';
import {
  getConfig,
  normalizeProductsList,
  normalizeProductsHistoryList,
} from './utils';
import { withNavigation } from 'react-navigation';
import { INVENTORY_STATUS } from './constants';
import { normalizeRecordTypes } from '../DashboardScreen/InventoriesWidget/utils';
import moment from 'moment';
import { INVENTORY_FORM_TYPE } from '../../constants/Inventories';

const initialValue = {
  editingType: INVENTORY_FORM_TYPE.preview,
  isLoading: true,
  preselectedProducts: [],
  products: [],
  productsHistory: [],
  error: null,
  config: {},
  lastInventoryCreatedDate: null,
  transactionDetails: [],
  recordType: {},
  formInitialValues: {},
};
export const InventoryContext = React.createContext(initialValue);

const inventoryKeys = {
  CreatedDate: 'createdDate',
  Id: 'id',
  OCE__Status__c: 'status',
  RecordTypeId: 'recordTypeId',
  OCE__Comments__c: 'comments',
  Name: 'name',
  OCE__Reason__c: 'reason',
  OCE__Auditor__c: 'auditor',
};

const configKeys = {
  OCE__SIShowSystemCalculatedFields__c: 'showCalculatedFields',
  OCE__SIShowSystemCount__c: 'showSystemCount',
  SetupOwnerId: 'ownerId',
  OCE__SIShowLoggedInUserRecords__c: 'record',
  OCE__STFinalStatus__c: 'status',
  OCE__SIHistoryListHidden__c: 'historyHidden',
};

const ContextProvider = ({ children, navigation }) => {
  const [state, setState] = useState(initialValue);
  const { userId, profileId, orgId } = useContext(AppContext);

  const handleFetchConfig = async () => {
    const [configsData] = await fetchConfigs(userId, profileId, orgId);

    const configs = normalizer(configKeys)(configsData);
    return getConfig(configs, [userId, profileId, orgId]);
  };

  const handleEditingType = editingType =>
    setState(prevState => ({ ...prevState, editingType }));

  useEffect(() => {
    handlePrepareData();
  }, []);

  const handlePrepareData = async () => {
    try {
      let inventory,
        transactionDetails,
        lastSubmittedInventoriesDetails,
        inventoryDetails,
        products,
        preselectedProducts,
        lastInventoryCreatedDate;
      const config = await handleFetchConfig();

      const [recordTypesData] = await fetchInventoryTypes();
      const recordTypes = normalizeRecordTypes(recordTypesData);

      const [lastInventories] = await fetchLastSubmittedInventory(
        userId,
        config.record
      );

      if (lastInventories.length) {
        inventory = [lastInventories[lastInventories.length - 1]];

        lastInventoryCreatedDate = inventory[0].CreatedDate;
        [lastSubmittedInventoriesDetails] = await fetchInventoryDetail(
          inventory[0].Id,
          true
        );
      } else {
        inventory = [
          {
            status: INVENTORY_STATUS.inProgress,
          },
        ];
      }

      if (navigation.state.params.id) {
        // preview or edit existing inventory
        [inventory] = await fetchInventoryById(navigation.state.params.id);
        [inventoryDetails] = await fetchInventoryDetail(
          navigation.state.params.id,
          state.editingType === INVENTORY_FORM_TYPE.preview
            ? false
            : inventory[0].OCE__Status__c !== INVENTORY_STATUS.inProgress
        );
      }

      [inventory] = normalizer(inventoryKeys)(inventory);

      if (inventory.status === INVENTORY_STATUS.inProgress) {
        [transactionDetails] = await fetchTransactionDetails(
          null,
          userId,
          config.status || 'Submitted'
        );
      } else {
        [transactionDetails] = await fetchTransactionDetails(
          navigation.state.params.id
        );
      }
      let productsHistory = normalizeProductsHistoryList(transactionDetails);

      const [productsList] = await fetchActiveLotsProducts();

      [products, preselectedProducts] = normalizeProductsList(
        productsList,
        lastSubmittedInventoriesDetails,
        inventoryDetails,
        transactionDetails,
        navigation.state.params.id ? inventory.status : null,
        Boolean(navigation.state.params.id)
      );

      setState(prevState => {
        let state = {
          ...prevState,
          isLoading: false,
          products,
          productsHistory,
          config,
          transactionDetails,
          lastInventoryCreatedDate,
        };
        if (navigation.state.params.id) {
          const recordType = recordTypes.all.find(
            el => el.Id === inventory.recordTypeId
          );

          state.editingType =
            navigation.state.params.type || INVENTORY_FORM_TYPE.preview;
          state.recordType = recordType;
          state.formInitialValues = {
            ...inventory,
            datetime: inventory.createdDate,
            products: preselectedProducts,
            buttonPressed: '',
          };
        } else {
          state.editingType = INVENTORY_FORM_TYPE.edit;
          state.recordType = navigation.state.params.recordType;
          state.formInitialValues = {
            id: '',
            datetime: moment(),
            products: preselectedProducts,
            status: INVENTORY_STATUS.inProgress,
            buttonPressed: '',
          };
        }

        return state;
      });
    } catch (error) {
      console.warn({ error });
      setState(prevState => ({
        ...prevState,
        isLoading: false,
        error,
      }));
    }
  };

  return (
    <InventoryContext.Provider
      value={{
        ...state,
        actions: { handleEditingType },
      }}
    >
      {children}
    </InventoryContext.Provider>
  );
};

export default withNavigation(ContextProvider);
