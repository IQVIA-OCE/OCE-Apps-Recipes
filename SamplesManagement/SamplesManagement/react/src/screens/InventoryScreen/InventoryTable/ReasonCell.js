import React, { useContext, useEffect } from 'react';
import { Select, Text } from 'apollo-react-native';
import { useField, useFormikContext } from 'formik';
import { InventoryContext } from '../InventoryContext';
import { INVENTORY_STATUS, REASON_LIST } from '../constants';
import { INVENTORY_FORM_TYPE } from '../../../constants/Inventories';
import { StyleSheet, View } from 'react-native';

const ReasonCell = ({ row }) => {
  const { values } = useFormikContext();
  const {
    editingType,
    config: { showCalculatedFields, showSystemCount },
  } = useContext(InventoryContext);

  const fieldIndex = values.products.findIndex(
    ({ lotNumberId }) => row.lotNumberId === lotNumberId
  );
  const [field, { touched, error }, helper] = useField(
    `products[${fieldIndex}]`
  );

  useEffect(() => {
    if (editingType !== INVENTORY_FORM_TYPE.preview) {
      if (parseInt(row.physicalQuantity) === parseInt(row.systemCount)) {
        helper.setValue({
          ...field.value,
          discrepancyReason: '',
          reasonObj: null,
        });
      }
      if (field.value.discrepancyReason && !field.value.reasonObj) {
        helper.setValue({
          ...field.value,
          reasonObj: REASON_LIST.find(
            el => el.id === field.value.discrepancyReason
          ),
        });
      }
    }
  }, [row.physicalQuantity]);

  // custom logic on specific configuration
  if (
    editingType === INVENTORY_FORM_TYPE.edit &&
    showCalculatedFields &&
    !showSystemCount &&
    values.status !== INVENTORY_STATUS.saved
  ) {
    return null;
  }

  if (
    (editingType === INVENTORY_FORM_TYPE.edit ||
      editingType === INVENTORY_FORM_TYPE.editSaved) &&
    !row.selected &&
    parseInt(row.physicalQuantity) === parseInt(row.systemCount)
  ) {
    return null;
  }

  if (editingType === INVENTORY_FORM_TYPE.preview) {
    return <Text>{field.value.discrepancyReason}</Text>;
  }

  return (
    <View style={styles.root}>
      <Select
        options={REASON_LIST}
        placeholder={'-None-'}
        value={field.value.reasonObj}
        onChange={value => {
          helper.setValue({
            ...field.value,
            discrepancyReason: value ? value.id : '',
            reasonObj: value,
          });
        }}
        fullWidth
        error={
          touched && error && error.discrepancyReason
            ? error.discrepancyReason
            : null
        }
        helperText={
          touched && error && error.discrepancyReason
            ? error.discrepancyReason
            : null
        }
      />
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    paddingVertical: 5,
    width: '100%',
  },
});

export default ReasonCell;
