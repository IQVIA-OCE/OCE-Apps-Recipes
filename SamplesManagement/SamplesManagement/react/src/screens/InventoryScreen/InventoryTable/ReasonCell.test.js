import React from 'react';
import ReasonCell from './ReasonCell';
import { Formik } from 'formik';
import { Select } from 'apollo-react-native';
import { InventoryContext } from '../InventoryContext';
import { INVENTORY_FORM_TYPE } from '../../../constants/Inventories';
import { render, act } from '@testing-library/react-native';

const Component = () => <ReasonCell row={{ id: '1' }} />;

describe('ReasonCell', () => {
  it('should render properly', async () => {
    const { getByTestId, container } = render(
      <InventoryContext.Provider
        value={{ editingType: INVENTORY_FORM_TYPE.edit, config: {} }}
      >
        <Formik initialValues={{ products: [{ id: '1', comments: '' }] }}>
          <Component />
        </Formik>
      </InventoryContext.Provider>
    );

    act(() => {
      container.findByType(Select).props.onChange({ id: '1', label: '1' });
    });

    expect(container.findByType(Select).props.value).toEqual({ id: '1', label: '1' });
    expect(getByTestId('ReasonCell')).toBeTruthy();
  });

  it('should NOT render component', async () => {
    const { queryByTestId, rerender } = render(
      <InventoryContext.Provider
        value={{
          config: { showCalculatedFields: true, showSystemCount: false },
          editingType: INVENTORY_FORM_TYPE.edit,
        }}
      >
        <Formik
          initialValues={{
            status: 'InProgress',
            products: [{ id: '1', comments: '' }],
          }}
        >
          <Component />
        </Formik>
      </InventoryContext.Provider>
    );

    expect(queryByTestId('ReasonCell')).toBeNull();
  });
});
