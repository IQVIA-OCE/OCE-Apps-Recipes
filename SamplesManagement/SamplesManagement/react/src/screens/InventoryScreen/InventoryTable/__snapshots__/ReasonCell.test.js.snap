// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReasonCell should NOT render component 1`] = `
<View
  style={
    Object {
      "paddingVertical": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View>
      <View
        collapsable={false}
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "backgroundColor": "#F6F6F6",
                "borderColor": "#d9d9d9",
                "borderRadius": 4,
                "borderWidth": 1,
                "elevation": 0,
                "height": 40,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "flexGrow": 1,
                "paddingLeft": 8,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 14,
                  "left": 14,
                  "right": 14,
                  "top": 14,
                }
              }
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "400",
                    },
                  ]
                }
              >
                -None-
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            centered={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            nativeID="animatedComponent"
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={Object {}}
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "margin": 0,
                    },
                  ],
                ]
              }
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ï…€
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`ReasonCell should NOT render component 2`] = `
<View
  style={
    Object {
      "paddingVertical": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View>
      <View
        collapsable={false}
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "backgroundColor": "#F6F6F6",
                "borderColor": "#d9d9d9",
                "borderRadius": 4,
                "borderWidth": 1,
                "elevation": 0,
                "height": 40,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "flexGrow": 1,
                "paddingLeft": 8,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 14,
                  "left": 14,
                  "right": 14,
                  "top": 14,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "400",
                    },
                  ]
                }
              >
                -None-
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            centered={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={Object {}}
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "margin": 0,
                    },
                  ],
                ]
              }
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ï…€
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`ReasonCell should render properly 1`] = `
<View
  style={
    Object {
      "paddingVertical": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View>
      <View
        collapsable={false}
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "backgroundColor": "#F6F6F6",
                "borderColor": "#d9d9d9",
                "borderRadius": 4,
                "borderWidth": 1,
                "elevation": 0,
                "height": 40,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "flex": 1,
                "flexDirection": "row",
                "flexGrow": 1,
                "paddingLeft": 8,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 14,
                  "left": 14,
                  "right": 14,
                  "top": 14,
                }
              }
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "flex": 1,
                  "opacity": 1,
                }
              }
            >
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": "400",
                    },
                  ]
                }
              >
                1
              </Text>
            </View>
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            centered={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            onClick={[Function]}
            onLayout={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            style={Object {}}
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                    },
                    Object {
                      "borderRadius": 18,
                      "height": 36,
                      "width": 36,
                    },
                    undefined,
                    Object {
                      "margin": 0,
                    },
                  ],
                ]
              }
            >
              <View>
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ï…€
                </Text>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`ReasonCell should render properly 2`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
        "textAlign": "left",
        "writingDirection": "ltr",
      },
      undefined,
    ]
  }
/>
`;
